spdx/document.py

```diff
@@ -366,58 +358,52 @@ def validate(self, messages):
         Validate all fields of the document and update the
         messages list with user friendly error messages for display.
         """
-        messages = self.validate_version(messages)
-        messages = self.validate_data_lics(messages)
-        messages = self.validate_name(messages)
-        messages = self.validate_spdx_id(messages)
-        messages = self.validate_namespace(messages)
-        messages = self.validate_ext_document_references(messages)
-        messages = self.validate_creation_info(messages)
-        messages = self.validate_packages(messages)
-        messages = self.validate_extracted_licenses(messages)
-        messages = self.validate_reviews(messages)
-        messages = self.validate_snippet(messages)
-        # messages = self.validate_annotations(messages)
-        # messages = self.validate_relationships(messages)
-
+        if isinstance(messages, list):
+            messages = ErrorMessages()
+
+        messages.push_context(self.name)
+        self.validate_version(messages)
+        self.validate_data_lics(messages)
+        self.validate_name(messages)
+        self.validate_spdx_id(messages)
+        self.validate_namespace(messages)
+        self.validate_ext_document_references(messages)
+        self.validate_creation_info(messages)
+        self.validate_packages(messages)
+        self.validate_extracted_licenses(messages)
+        self.validate_reviews(messages)
+        self.validate_snippet(messages)
+        # self.validate_annotations(messages)
+        # self.validate_relationships(messages)
```

Comment:
Let's get rid of dead commented code

Reply:
well, I think it does not make much sense to have it commented out. was introduced here: https://github.com/spdx/tools-python/commit/209a92563ea321403e0e8abc2308644367d73996